docs/design/ovs-pipeline.md

```diff
@@ -1,1197 +1,1852 @@
+from remote Pods through a tunnel). The packets are not matched by any flows 1-6, which means that they are here
+unexpectedly and should be dropped.
+
+Flow 8 is the table-miss flow, which matches "tracked" and non-new packets from Egress connections and forwards
+them to table [L2ForwardingCalc]. `ToGatewayRegMark` is also loaded for these packets.
+
+### L3DecTTL
+
+This is the table to decrement TTL for IP packets.
+
+If you dump the flows of this table, you may see the following:
 
 ```text
-table=71, priority=200,ct_state=+new+trk,ip,in_port="pod2-357c21" actions=mod_dl_src:e2:e5:a4:9b:1c:b1,mod_dl_dst:aa:bb:cc:dd:ee:ff,load:0x1->NXM_NX_REG1[],set_field:0x10000/0x10000->reg0,load:0xc0a80a66->NXM_NX_TUN_IPV4_DST[],goto_table:72
+1. table=L3DecTTL, priority=210,ip,reg0=0x2/0xf actions=goto_table:SNATMark
+2. table=L3DecTTL, priority=200,ip actions=dec_ttl,goto_table:SNATMark
+3. table=L3DecTTL, priority=0 actions=goto_table:SNATMark
 ```
 
-Last, when a SNAT IP configured for Egresses is on the local Node, an additional
-flow is added in [SNATTable] for egress traffic from remote Node that should
-use the SNAT IP. The flow matches the tunnel destination IP (which should be
-equal to the SNAT IP), and sets the 8 bits ID of the SNAT IP to pkt_mark.
+Flow 1 matches packets with `FromGatewayRegMark`, which means that these packets enter the OVS pipeline from the local
+Antrea gateway, as the host IP stack should have decremented the TTL already for such packets, TTL should not be
+decremented again.
+
+Flow 2 is to decrement TTL for packets which are not matched by flow 1.
+
+Flow 3 is the table-miss flow that should remain unused.
+
+### SNATMark
+
+This table marks connections requiring SNAT within the OVS pipeline, distinct from Egress SNAT handled by iptables.
+
+If you dump the flows of this table, you may see the following:
 
 ```text
-table=71, priority=200,ct_state=+new+trk,ip,tun_dst="192.168.10.101" actions=set_field:0x1/0xff->pkt_mark,goto_table:80
+1. table=SNATMark, priority=200,ct_state=+new+trk,ip,reg0=0x22/0xff actions=ct(commit,table=SNAT,zone=65520,exec(set_field:0x20/0x20->ct_mark,set_field:0x40/0x40->ct_mark))
+2. table=SNATMark, priority=200,ct_state=+new+trk,ip,reg0=0x12/0xff,reg4=0x200000/0x2200000 actions=ct(commit,table=SNAT,zone=65520,exec(set_field:0x20/0x20->ct_mark))
+3. table=SNATMark, priority=190,ct_state=+new+trk,ip,nw_src=10.10.0.23,nw_dst=10.10.0.23 actions=ct(commit,table=SNAT,zone=65520,exec(set_field:0x20/0x20->ct_mark,set_field:0x40/0x40->ct_mark))
+4. table=SNATMark, priority=190,ct_state=+new+trk,ip,nw_src=10.10.0.24,nw_dst=10.10.0.24 actions=ct(commit,table=SNAT,zone=65520,exec(set_field:0x20/0x20->ct_mark,set_field:0x40/0x40->ct_mark))
+5. table=SNATMark, priority=0 actions=goto_table:SNAT
 ```
 
-### L3DecTTLTable (72)
+Flow 1 matches the first packet of hairpin Service connections, identified by `FromGatewayRegMark` and `ToGatewayRegMark`,
```

Comment:
Yes, it should be specific to `proxyAll`. The use cases are:

- encap mode,  a client from the external network or the local Node, destines for a Service, and the Endpoint is a hostNetwork Pod or an external link.
- noEncap mode,  a client from the external network or the local Node, destines for a Service, and the Endpoint is a hostNetwork Pod, a remote non-hostNetwork Pod or an external link.

Reply:
Ok, we should mention that it is required for `proxyAll` in the flow description.

Do you know why the flow is installed even when proxyAll is disabled? Is it for simplicity and because it is harmless?

Reply:
> Ok, we should mention that it is required for proxyAll in the flow description

Will add.

> Do you know why the flow is installed even when proxyAll is disabled? Is it for simplicity and because it is harmless?

Yes, just for simplicity and it is harmless when `proxyAll` is not enabled.

Reply:
Flow 2 should be also specific to `proxyAll`.